src/python/py/models/builder.py

```diff
@@ -1164,13 +1180,23 @@ def make_common_mask_reformat_subgraph(self, basename, root_input, unsqueeze_for
 class LlamaModel(Model):
     def __init__(self, config, io_dtype, onnx_dtype, ep, cache_dir, extra_options):
         super().__init__(config, io_dtype, onnx_dtype, ep, cache_dir, extra_options)
-        self.model_inputs = ["input_ids", "attention_mask", "position_ids"]
+        if self.enable_cuda_graph:
+            self.model_inputs = ["input_ids", "position_ids", "seqlens_k", "total_seq_len"]
+        else:
+            self.model_inputs = ["input_ids", "attention_mask", "position_ids"]
 
     def make_attention_mask_reformatting(self):
         if not self.attention_attrs["use_gqa"]:
             super().make_attention_mask_reformatting()
             return
 
+        if self.enable_cuda_graph:
+            # ORT does not allow nodes to be placed on mulitple execution providers 
+            # with cuda graph enabled. Thus the attention mask is deprecated and the
```

Comment:
Should we consider having `seqlens_k` and `total_seq_len` as raw inputs by default instead of `attention_mask` when GQA is used?

Reply:
Yes using seqlens_k and total_seq_len by default for GQA is the goal. But at the moment I'm concerning if the change will affect some customers that already have the inference code that supports masks so just keep both options